src/field.cpp

```diff
@@ -0,0 +1,34 @@
+#include "field.h"
+
+typedef std::mt19937 rng_t;
+
+Field::Field(sf::RenderWindow& w):
```

Comment:
I’m generally not a huge fan of passing in a reference and storing the reference in an ivar. What if you got a reference to the renderwindow, which was allocated on the stack? Then if the lifecycle of your field object was longer than the renderwindow, the corresponding ivar would be referring to an in unallocated variable. Since you’re a c++ big boy now, why not look into shared pointers for holding the renderwindow?

Reply:
The render window is going to be initialized in the game class and passed down so field won't outlive window because field will die before game.

Reply:
The use of shared pointers to store ivars is more of a defensive programming practice so that’s why I recommended it.